framework/src/modules/reward/endpoint.ts

```diff
@@ -15,17 +15,17 @@
 import { ModuleEndpointContext } from '../../types';
 import { BaseEndpoint } from '../base_endpoint';
 import { calculateDefaultReward } from './calculate_reward';
-import { DefaultReward, EndpointInitArgs } from './types';
+import { DefaultReward, ModuleConfig } from './types';
+
+export interface EndpointInitArgs {
+	config: ModuleConfig;
```

Comment:
This `EndpointInitArgs` interface seems completely unnecessary. It doesn't add much value, as it only wraps `ModuleConfig` interface, without providing anything new.

How about we get rid of that boilerplate, and pass `ModuleConfig` object to `RewardEndpoint.init()`. That is how I implemented it recently in `TokenEndpoint`:

```typescript
public init(moduleConfig: ModuleConfig) {
    this._moduleConfig = moduleConfig;
}
```